src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -165,6 +166,17 @@ SELECT COUNT(r.id)
             """)
     List<Long> countNumberOfFinishedAssessmentsByExerciseIdIgnoreTestRuns(@Param("exerciseId") Long exerciseId);
 
+    @Query("""
+            SELECT r
+                FROM StudentParticipation p join p.submissions s join s.results r
```

Comment:
until now we were using LEFT JOIN. you might want to adjust this query?

```suggestion
                FROM StudentParticipation p LEFT JOIN p.submissions s LEFT JOIN s.results r
```

Reply:
I'm not sure 100% but I think in this case it is better to use `join`, and not the `left join`, as we only are interested in the `result r`.